python/RobotControl/RunningWithSSH.py

```diff
@@ -16,27 +25,112 @@ def run_script_on_robot(script: str) -> str:
         Returns: 
             An error message or an empty string.
     """
-    robot.interpreter_mode.start()
+    robot.ssh.write_script(script)
+    robot.controller.load_program()
+    robot.controller.start_program()
+    sleep(0.1)
 
-    interpreter_script = "halt\n" + script
-    result = robot.interpreter_mode.send_small_command_on_interpreter(interpreter_script)
-    non_recurring_logger.debug(f"Result of sending script to interpreter: {result}")
+    latest_errors = robot.ssh.read_lines_from_log(2)
 
-    error_message = result.lower().split(":")[1]
-    out = None
+    if "Compile error" in latest_errors or "Lexer exception" in latest_errors:
+        non_recurring_logger.debug("Compile error or Lexer exception found")
+        error = latest_errors.split("\n")[1]
+        parts = re.split(r'ERROR\s+-', error, maxsplit=1)
+        return parts[1]
```

Comment:
There was a time where this returned an error because there was only 1 part. Has that been fixed at this point?

Reply:
Yas queeeen!